common/models/model-definition.js

```diff
@@ -182,8 +182,8 @@ function ready(ModelDefinition) {
     for (var ix in relatedData) {
       var data = Entity.getConfigFromData(relatedData[ix]);
       delete data[relation.foreignKey];
-      delete data[relation.embed.key];
-      delete data[relation.embed.keyFn];
+      delete data.name;
+      delete data[relation.embed.keySetter];
```

Comment:
The idea is to keep `cleanRelatedData` totally not opinionated, just providing extension hooks allowing domain entities to implement whatever custom rules they need.